features/filiatie.feature

```diff
@@ -0,0 +1,470 @@
+# language: nl
+
+Functionaliteit: Als gemeente wil ik inzicht in de filiatie van Kadastraal onroerende zaken
```

Comment:
Hier wordt gesproken over Kadastraal onroerende zaken, maar in de rest van de feature wordt alleen over perceel en appartementsrecht gesproken. Ook in de API kom je geen Perceel en Appartementsrecht resource tegen. Dit kan verwarrend zijn. Ik denk dat het handig is om een glossary te maken om al deze termen te beschrijven en die dan te refereren

Reply:
Het is volgens mij niet handig om de feature zoals het nu is op te stellen. Je moet heen en weer scrollen om de Givens voor een scenario te kunnen lezen. Voor mij is dit alleen goed te volgen door het bestand twee keer te openen en naast elkaar te leggen

Reply:
@MelvLee ik gebruik nu per rule dezelfde dataset, dus dezelfde dataset in meerdere scenario's. Voor zover ik weet moet je Gegeven opnemen in Achtergrond of per scenario. Weet jij of (en hoe) het mogelijk is Gegeven te definieren per Rule?
Of zou jij het dan voor elk scenario herhalen?

Reply:
Zoals ik het **NU** begrijp zou ik deze feature splitsen in meerdere feature bestanden. Eén voor percelen en één voor appartementsrechten. En mogelijk ook voor percelen die overgaan in appartementsrechten? Kan dit ook omgekeerd gebeuren, dus appartementsrechten die overgaan in percelen?

Reply:
Ik weet niet of het handig is om één dataset te maken en die te hergebruiken. Het wordt heel erg puzzelen als je straks nieuwe scenario's toevoegt of verwijderd. Op een gegeven moment weet je niet meer wat je in de dataset kan aanpassen (breek je daardoor niet een oude scenario?) of moet verwijderen. Je krijgt dan hetzelfde wat ik in code ken, nl. geen code meer opruimen omdat je niet meer kan afleiden of een andere stukje code deze indirect gebruikt.

Reply:
> Zoals ik het **NU** begrijp zou ik deze feature splitsen in meerdere feature bestanden. Eén voor percelen en één voor appartementsrechten. En mogelijk ook voor percelen die overgaan in appartementsrechten? Kan dit ook omgekeerd gebeuren, dus appartementsrechten die overgaan in percelen?

Volgens mij is filiatie een concept waar drie aspecten aan zitten. Ik zou dit wel in 1 feature-file beschrijven zoals Frank dat nu gedaan heeft. Met de rules die beschreven zijn dek je alle scenario'saf mijns inziens.

Reply:
> Ik weet niet of het handig is om één dataset te maken en die te hergebruiken. Het wordt heel erg puzzelen als je straks nieuwe scenario's toevoegt of verwijderd. Op een gegeven moment weet je niet meer wat je in de dataset kan aanpassen (breek je daardoor niet een oude scenario?) of moet verwijderen. Je krijgt dan hetzelfde wat ik in code ken, nl. geen code meer opruimen omdat je niet meer kan afleiden of een andere stukje code deze indirect gebruikt.

Wat zou jouw alternatieve voorstel dan zijn ? Verschillen de isOntstaanUit en de isOvergegaanIn dan ernstig van structuur ?

Reply:
Ik vind het prima om het bij één bestand te houden. Het belangrijkste is dat jullie het nog moeten snappen als er in de toekomst vragen over wordt gesteld.

Reply:
Als ik dit zo lees denk ik dat het kadaster zelf een beeld heeft van wat zij exact gaan leveren, het is immers hun domein. Het probleem lijkt hier te zijn dat het onzeker is of dit een gedeeld beeld is. Of wij als gebruiker zeker weten dat wij (consumer en provider) het over hetzelfde hebben. De costumer zero heeft hierover veel vragen en geeft dat aan. En hij doet een voorstel over hoe hij het misschien wel kan snappen. Een ontwikkelaar redeneert vanuit code en objecten. Wij laten dat in de OAS los, maar dat moet wel weer terug worden gemapt naar kleine stukjes. Wij kunnen toch aansluiten bij de belevingswereld van onze klant door de documentatie daarop aan te passen?

@MelvLee probeer jij in jouw stukje over hergebruik te zeggen dat er teveel afhankelijkheden zijn in dit geval (en jij het overzicht kwijt raakt) en het daarom nu wel nodig is om te splitsen en als gevolg daarvan te dupliceren? Dus een feature met een hoofdstuk voor perceel en hoofdstuk voor appartementsrecht? Het doel is hier: de klant inzicht geven door het gedrag te beschrijven (bijvoorbeeld wanneer is een appartementsrechtvolgnummer gevuld etc.)

@MelvLee zou jij nog even commentaar kunnen geven op de vraag van Frank bovenaan?

Reply:
> 
> 
> @MelvLee ik gebruik nu per rule dezelfde dataset, dus dezelfde dataset in meerdere scenario's. Voor zover ik weet moet je Gegeven opnemen in Achtergrond of per scenario. Weet jij of (en hoe) het mogelijk is Gegeven te definieren per Rule?
> Of zou jij het dan voor elk scenario herhalen?

@fsamwel, als je per rule dezelfde dataset gebruik, dan zou ik voor elke rule een eigen feature file maken. Het is inderdaad niet mogelijk om een aparte Achtergrond voor een rule te definieren